src/pineko/check.py

```diff
@@ -92,31 +93,100 @@ def is_fonll_b(fns, lumi):
     return False
 
 
-def contains_fact(grid):
-    """Check whether factorization scale-variations are available in the pineappl grid.
+def contains_fact(grid, max_as, max_al):
+    """Check whether factorization scale-variations are available in the pineappl grid for the requested order.
 
     Parameters
     ----------
         grid: pineappl.grid.Grid
             Pineappl grid
+        max_as: int
+            max as order
+        max_al: int
+            max al order
+    Returns
+    -------
+        : bool
+        is fact scale-variation available for as
+        : bool
+        is fact scale-variation available for al
     """
-    order_list = [order.as_tuple() for order in grid.orders()]
+    order_array = np.array([order.as_tuple() for order in grid.orders()])
+    order_mask = pineappl.grid.Order.create_mask(grid.orders(), max_as, max_al)
+    order_list = order_array[order_mask]
+    as_orders = []
+    al_orders = []
```

Comment:
I like the idea of providing both `as` and `al` - however there is too much repetition here ... and the check for both is essentially the same ...

Reply:
Do you mean that maybe we need a separate function to do the checks? Or do you have something else in mind? 
Because of course is repetitive but they are not exactly the same. Moreover, I don't believe that splitting the check on `as` and the check on `al` would be worth.

Reply:
yes, I thought about defining an (inline) function to be called twice ... however, I agree it is a matter of taste, so you can either decide yourself or ask for a third opinion, e.g. @AleCandido